kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/WatchTest.java

```diff
@@ -58,38 +58,92 @@
 
+  private static final ObjectMapper mapper = new ObjectMapper();
+
   @Test
-  public void testDeletedAndOutdated() throws InterruptedException {
+  public void testDeletedOutdatedAndAdded() throws InterruptedException {
     logger.info("testDeletedAndOutdated");
     KubernetesClient client = server.getClient().inNamespace("test");
 
+    final String path = "/api/v1/namespaces/test/pods?fieldSelector=metadata.name%3Dpod1&resourceVersion=1&watch=true";
+    final String pathWithoutResourceVersion = "/api/v1/namespaces/test/pods?fieldSelector=metadata.name%3Dpod1&watch=true";
+
     // DELETED event, then history outdated
     server.expect()
-        .withPath("/api/v1/namespaces/test/pods?fieldSelector=metadata.name%3Dpod1&resourceVersion=1&watch=true")
+        .withPath(path)
         .andUpgradeToWebSocket().open().waitFor(2000).andEmit(new WatchEvent(pod1, "DELETED")).waitFor(2000)
         .andEmit(outdatedEvent).done().once();
 
+    // ADDED event
+    server.expect()
+        .withPath(pathWithoutResourceVersion)
+        .andUpgradeToWebSocket().open().waitFor(2000).andEmit(new WatchEvent(pod1, "ADDED")).done().once();
+
     final CountDownLatch deleteLatch = new CountDownLatch(1);
-    final CountDownLatch closeLatch = new CountDownLatch(1);
+    final CountDownLatch addLatch = new CountDownLatch(1);
+    final boolean[] onCloseCalled = {false};
     try (Watch watch = client.pods().withName("pod1").withResourceVersion("1").watch(new Watcher<Pod>() {
       @Override
       public void eventReceived(Action action, Pod resource) {
         switch (action) {
         case DELETED:
           deleteLatch.countDown();
           break;
+        case ADDED:
+          addLatch.countDown();
+          break;
         default:
           throw new AssertionFailedError();
         }
       }
 
       @Override
       public void onClose(KubernetesClientException cause) {
-        closeLatch.countDown();
+        onCloseCalled[0] =true;
       }
     })) /* autoclose */ {
       assertTrue(deleteLatch.await(10, TimeUnit.SECONDS));
-      assertTrue(closeLatch.await(10, TimeUnit.SECONDS));
+      assertTrue(addLatch.await(10, TimeUnit.SECONDS));
+    }
+    assertTrue(onCloseCalled[0]);
+  }
+
+  @Test
+  public void testHttpOutdatedAndModified() throws InterruptedException,
+                                                   JsonProcessingException {
+    KubernetesClient client = server.getClient().inNamespace("test");
+
+    final String path = "/api/v1/namespaces/test/pods?fieldSelector=metadata.name%3Dpod1&resourceVersion=1&watch=true";
+    final String pathWithoutResourceVersion = "/api/v1/namespaces/test/pods?fieldSelector=metadata.name%3Dpod1&watch=true";
+
+    server.expect()
+        .withPath(path)
+        .andReturn(200, new StatusBuilder().withCode(200).build()).once();
+    server.expect()
+        .withPath(path)
+        .andReturn(200, mapper.writeValueAsString(outdatedEvent) + '\n').once();
```

Comment:
I couldn't figure out a better way to do this. The mock server requires a strict line with line feed.

Reply:
Also I'm not entirely sure HTTP code 200 with 410 in the body is really what k8s returns.